SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp

```diff
@@ -298,5 +446,210 @@ int GenerateTypeBindingSchema(FCodeWriter& Writer, int ComponentId, UClass* Clas
 		UE_LOG(LogTemp, Warning, TEXT("Unreal GDK currently does not support Reliable Multicast RPCs. These RPC will be treated as unreliable:\n%s"), *AllReliableMulticasts);
 	}
 
+	ClassToSchema.Add(Class, ActorSchemaData);
+
+	Writer.WriteToFile(FString::Printf(TEXT("%s%s.schema"), *SchemaPath, *UnrealNameToSchemaTypeName(Class->GetName())));
+
 	return IdGenerator.GetNumUsedIds();
 }
+
+FSubobjectSchemaData GenerateSubobjectSpecificSchema(FCodeWriter& Writer, FComponentIdGenerator& IdGenerator, FString PropertyName, TSharedPtr<FUnrealType>& TypeInfo, UClass* ComponentClass)
+{
+	FUnrealFlatRepData RepData = GetFlatRepData(TypeInfo);
+
+	FSubobjectSchemaData SubobjectData;
+	SubobjectData.Class = ComponentClass;
+
+	for (EReplicatedPropertyGroup Group : GetAllReplicatedPropertyGroups())
+	{
+		if (RepData[Group].Num() == 0)
+		{
+			continue;
+		}
+
+		Writer.PrintNewLine();
+
+		FString ComponentName = PropertyName + GetReplicatedPropertyGroupName(Group);
+		Writer.Printf("component {0} {", *ComponentName);
+		Writer.Indent();
+		Writer.Printf("id = {0};", IdGenerator.GetNextAvailableId());
+		Writer.Printf("data {0};", *SchemaReplicatedDataName(Group, ComponentClass));
+		Writer.Outdent().Print("}");
+
+		if (Group == REP_MultiClient)
+		{
+			SubobjectData.SchemaComponents[ESchemaComponentType::TYPE_Data] = IdGenerator.GetCurrentId();
+		}
+		else if (Group == REP_SingleClient)
+		{
+			SubobjectData.SchemaComponents[ESchemaComponentType::TYPE_OwnerOnly] = IdGenerator.GetCurrentId();
+		}
```

Comment:
Nit: add
```
else
{
	checkNoEntry();
}
```
so it's easier to find places to change if we add a new replicated property group.